python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -289,6 +290,61 @@ def default_blocksize(scheduler: str) -> int:
     return min(max(blocksize, 1_000_000_000), 10_000_000_000)
 
 
+@dataclasses.dataclass(frozen=True)
+class StatsPlanningOptions:
```

Comment:
These new "nested" config options are probably the most important part of this PR.